File: primeChecker.c
#include<stdio.h>
#include<math.h>
int isprime(int n){
    for(int i=2; i< n;  i++){
        if(n%i ==0){
            return 0;
        }
    }
    return 1;
}
int main(){
    int n;

    printf("Enter the  value of n :");
    scanf("%d",&n);

    if(isprime(n)){
        printf("%d is a prime number\n",n);
    }else{
        printf("%d is  not a prime number \n",n);
    }
    
    printf("Prime numbers upto %d: ",n);
    for(int i=2; i<n; i++){
        if(isprime(i)){
            printf("%d ",i);
        }
    }

    return 0;
}

File: primeChecker.f90
function isPrime(n) result(isP)
    integer, intent(in):: n
    integer :: i,isP

    isP = 1

    do i=2, n-1
        if (mod(n,i) == 0) then 
            isP = 0
            exit 
        end if
    end do

end function isPrime

program primeChecker 
    integer  :: n

    write(*,*) "Enter any number : "
    read (*,*) n

    if (isPrime(n)==1) then
        write(*,*) n, "is a prime number "
    else 
        write(*,*) n, "is not a prime number "
    end if


    write(*,*) "prime numbbers upto",n,"are"
    do i=2, n
        if(isPrime(i)==1) then  
            write(*,*) i
        end if
    end do
    
end program primeChecker
